### ISSUE TITLE
📎 Port `vue/no-unused-emit-declarations` from `eslint-plugin-vue`

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#6306]
### Description

Rule Source: https://eslint.vuejs.org/rules/no-unused-emit-declarations.html

Name suggestions:
- `noUnusedEmitDeclarations`

This should go in the Vue domain.

**Want to contribute?** Let us know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

**Comment by CDGardner** [comment]
Think I'd like to take a stab at this!

**Comment by sam-parsons** [comment]
This is an interesting issue - Biome's HTML/template analyzer and JS/script analyzer run independently on separate blocks of a .vue file from what I can tell — there is no mechanism for a JS lint rule to see `$emit()` calls that occur in the <template> block. This means any implementation of noUnusedEmitDeclarations in biome_js_analyze would false-positive on components where an emit is only called from the template, which is a common pattern:

```
<template>
  <button @click="$emit('close')">X</button>
</template>

<script setup>
const emit = defineEmits(['close'])
// 'close' is never called in the script — rule would incorrectly report it as unused
</script>
```

Couple questions (I'm new round these parts):

- Is there a way to access template AST from a JS lint rule that I'm missing?
- If not, would an implementation that only checks script-side emit calls be acceptable for now, with the limitation clearly documented in the rule's rustdoc, maybe a follow-up issue?

**Comment by dyc3** [comment]
Hmm, yeah I forgot about $emit when I labeled this a good first issue. It's possible but it's non trivial.

**Comment by sam-parsons** [comment]
I am interested in doing this issue if nobody else wants, but wanted to talk aloud about how to solve this.

Looking at the existing service layer (`EmbeddedValueReferences`, `EmbeddedExportedBindings`), it seems like the right approach is a new service in that same layer — one that walks template expressions and collects literal values passed to function calls, then injects them into the JS rule at analysis time. That way the JS rule can check declared emits against both script-side calls and template-side calls.

The question before building it: what level of generalization makes sense? Initial instinct was something string-specific (`EmbeddedStringCallArguments`), but since any literal — numbers, booleans, even object literals with primitive values — is statically knowable at parse time, something like `EmbeddedLiteralCallArguments` might be more appropriate and reusable for future rules. Is that the right direction, or is there probably another way of approaching this?

**Comment by dyc3** [comment]
Well, long term the solution we want to have is a vue-specific semantic model service. The current approach conflates vue/svelte/astro all into one service, but they do things differently enough that its clear the `EmbeddedValueReferences`/`EmbeddedExportedBindings` approach isn't the right way to do it.

So the next steps for this is to determine:
- what to collect in the vue semantic model
- what apis to expose to lint rules

I've created #10211 to discuss this further.
